apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -179,6 +179,8 @@ def read(
                     lo_hi = util.slice_to_range(ids)
                     if lo_hi is not None:
                         sr.set_dim_ranges(SOMA_ROWID, [lo_hi])
+                else:
```

Comment:
for any new code & behavior, please add a unit test to cover it.  In this case, to ensure the error ir raised as expected.